src/charm.py

```diff
@@ -202,85 +225,102 @@ def reconfigure_nats(self):
+                }
+            )
+            self.client._set_tls_ca(
+                self.ca_client.ca_certificate.public_bytes(
+                    encoding=serialization.Encoding.PEM
+                ).decode("utf-8")
+            )
 
         if self.nrpe_client.is_available:
             check_name = "check_{}".format(self.model.unit.name.replace("/", "_"))
-            self.nrpe_client.add_check(command=[
-                '/usr/lib/nagios/plugins/check_tcp',
-                '-H', str(self.client.listen_address),
-                '-p', str(self.model.config['client-port'])
-            ], name=check_name)
+            self.nrpe_client.add_check(
+                command=[
+                    "/usr/lib/nagios/plugins/check_tcp",
+                    "-H",
+                    str(self.client.listen_address),
+                    "-p",
+                    str(self.model.config["client-port"]),
+                ],
+                name=check_name,
+            )
             self.nrpe_client.commit()
 
-        tenv = Environment(loader=FileSystemLoader('templates'))
-        template = tenv.get_template('nats.cfg.j2')
+        tenv = Environment(loader=FileSystemLoader("templates"))
+        template = tenv.get_template("nats.cfg.j2")
         rendered_content = template.render(ctxt)
-        content_hash = self.generate_content_hash(rendered_content)
+        content_hash = self._generate_content_hash(rendered_content)
         old_hash = self.state.nats_config_hash
         if old_hash != content_hash:
-            logging.info(f'Config has changed - re-rendering a template to {self.NATS_SERVER_CONFIG_PATH}')
+            logging.info(
+                f"Config has changed - re-rendering a template to {self.NATS_SERVER_CONFIG_PATH}"
+            )
             self.state.nats_config_hash = content_hash
             self.NATS_SERVER_CONFIG_PATH.write_text(rendered_content)
             if self.state.is_started:
-                subprocess.check_call(['systemctl', 'restart', self.NATS_SERVICE])
+                subprocess.check_call(["systemctl", "restart", self.NATS_SERVICE])
         self.client.expose_nats(auth_token=self.state.auth_token)
 
-        client_port = self.model.config['client-port']
-        if (client_port is None or client_port == 0) and (self.state.client_port is not None or len(self.state.client_port) > 0):
+        client_port = self.model.config["client-port"]
+        if (client_port is None or client_port == 0) and (
+            self.state.client_port is not None or len(self.state.client_port) > 0
+        ):
             self._close_port(self.state.client_port)
         else:
-            port = '{}/tcp'.format(client_port)
+            port = "{}/tcp".format(client_port)
             if self.state.client_port is not None and port != self.state.client_port:
                 self._close_port(self.state.client_port)
             self._open_port(port)
```

Comment:
`ops` has it's own [`open_port`](https://ops.readthedocs.io/en/latest/#ops.Unit.open_port) method you can use.